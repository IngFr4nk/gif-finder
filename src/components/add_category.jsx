import { useState } from "react"
import PropTypes from 'prop-types'

export const Add_category = ({ placeholder, on_new_category} ) => {

    const [input_value, SetInput_value] = useState('')

    const change_input = ({target}) => {        
        SetInput_value(target.value)
    }

    const on_submit = (event) => {
       event.preventDefault()       
    }

    const on_add_categories = () => {    
        if(input_value.trim().length <= 1) return
        // addCategories(categories => [input_value, ...categories])
        SetInput_value('')
        on_new_category(input_value.trim())
    }

    return (
        <form onSubmit={event => on_submit(event)} aria-label='form'>
            <input
                type='text'
                placeholder={placeholder}
                value={input_value}
                onChange={change_input}
            />
            <button onClick={on_add_categories}>Find</button>
        </form>
    )
}

Add_category.propTypes = {
    on_new_category: PropTypes.func.isRequired
}