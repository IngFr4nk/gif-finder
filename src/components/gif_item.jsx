import PropsTypes from 'prop-types'

export const Gif_item = ({title, url}) => {
    return (
        <li className="card Gif">
            <img src={url} alt={title} />
            <span>{title}</span>                                            
        </li>
    )
}

Gif_item.propTypes = {
    title:  PropsTypes.string.isRequired,
    url: PropsTypes.string.isRequired
}