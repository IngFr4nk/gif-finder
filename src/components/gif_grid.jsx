import PropTypes from 'prop-types'
import useFetchImg from "../hooks/use_fetch_img"
import {Gif_item} from "../components"

export const Gif_grid = ({ category }) => {
    
    const {images} = useFetchImg(category)

    return (
        <>
            <h2>{category}</h2>
            <ol className="card-grid">
                {
                    images.map(({ id, title, url }) => (
                        <Gif_item key={id} title={title} url={url} />                       
                    ))
                }
            </ol>
        </>
    )
}

Gif_grid.propTypes = {
    catergoy: PropTypes.string.isRequired
}