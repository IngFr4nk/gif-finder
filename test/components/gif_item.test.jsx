import {render, screen} from '@testing-library/react'
import { Gif_item } from '../../src/components/gif_item'

describe('Test gif item component', () => {

    const title = 'hello bro!'

    const url   = 'https://google.com.co/'

    test('Deber hacer match con el snapshot', () => {
        const {container} = render(<Gif_item title={title} url={url} />)
        expect(container).toMatchSnapshot()
    })

    test('Debe mostrar la imagen con el url y el ALT indicado', () => {
        render(<Gif_item title={title} url={url} />)
        const {alt, src} = screen.getByRole('img')
        expect(alt).toBe(title)
        expect(src).toBe(url)
    })

    test('Deber mostrar titulo en el componente', () => {
        render(<Gif_item title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy()
    })
})