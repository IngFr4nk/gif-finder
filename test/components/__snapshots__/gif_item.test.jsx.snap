// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test gif item component Deber hacer match con el snapshot 1`] = `
<div>
  <li
    class="card Gif"
  >
    <img
      alt="hello bro!"
      src="https://google.com.co/"
    />
    <span>
      hello bro!
    </span>
  </li>
</div>
`;
